imap/util.c

```diff
@@ -335,6 +336,7 @@ int imap_parse_path(const char *path, struct ImapMbox *mx)
     if (url.scheme == U_IMAPS)
       mx->account.flags |= MUTT_ACCT_SSL;
 
+    url_free(&url);
     FREE(&c);
   }
   /* old PINE-compatibility code */
```

Comment:
yuou are leaking url in the `else` clause following. Admittedly, this is a non-problem, since I don't expect a query string in an imap uri. Anyway, it would be better to call `url_free` anyway...